pubsub.go

```diff
@@ -43,22 +51,116 @@ type ReceiveSettings struct {
 	Settings pubsub.ReceiveSettings
 }
 
+// SubscriptionSettings is an extension of Google PubSub's SubscriptionConfig that
+// enables further configuration for a subscription of a topic
+type SubscriptionSettings struct {
+	EnableMessageOrdering bool
+	RetainAckedMessages   bool
+}
+
+// TopicSettings is an extension of Google PubSub's TopicConfig that
+// enables further configuration for a topic
+type TopicSettings struct {
+	RetentionDurationInDays int
+}
+
 // NewPubSub creates a new PubSub client with the provided Config.
 func NewPubSub(c Config) (*PubSub, error) {
 	client, err := newClient(c.ProjectID, c.IsLocal, c.ServiceAccountFilePath)
 	if err != nil {
 		return nil, err
 	}
 
+	ts := TopicSettings{
```

Comment:
By making `Config` take in `TopicSettings` and `SubscriptionSettings` you won't have to assign it for lines 74 and 77